From cddb925ede6c5a3f578f55ea2184f731d0e039dd Mon Sep 17 00:00:00 2001
From: aiaimimi0920 <153103332+aiaimimi0920@users.noreply.github.com>
Date: Thu, 7 Mar 2024 05:19:24 +0800
Subject: [PATCH] #cache code

---
 scene/theme/default_theme.cpp | 58 +++++++++++++++--------------------
 1 file changed, 25 insertions(+), 33 deletions(-)

diff --git a/scene/theme/default_theme.cpp b/scene/theme/default_theme.cpp
index aaf715749d2e332fa0fe0f3ccad2fecb0132afec..a37913d422fdd8ab3ab7d9d98a67eb35c3a47915 100644
--- a/scene/theme/default_theme.cpp
+++ b/scene/theme/default_theme.cpp
@@ -166,26 +166,17 @@ void fill_default_theme(Ref<Theme> &theme, const Ref<Font> &default_font, const
 
 	// Panel
 
-	theme->set_stylebox("panel", "Panel", make_color_role_flat_stylebox(ColorRole::PRIMARY, style_normal_color, 0, 0, 0, 0));
+	theme->set_stylebox("panel", "Panel", make_color_role_flat_stylebox(ColorRole::PRIMARY_CONTAINER, Color(1, 1, 1, 0.6), 0, 0, 0, 0));
 
 	// ColorRect
 	theme->set_color_scheme("default_color_scheme", "ColorRect", default_color_scheme);
 	
 	// Button
-
-	// const Ref<StyleBoxFlat> button_normal = make_flat_stylebox(style_normal_color);
-	// const Ref<StyleBoxFlat> button_hover = make_flat_stylebox(style_hover_color);
-	// const Ref<StyleBoxFlat> button_pressed = make_flat_stylebox(style_pressed_color);
-	// const Ref<StyleBoxFlat> button_disabled = make_flat_stylebox(style_disabled_color);
-
-
-	const Ref<StyleBoxFlat> button_normal = make_color_role_flat_stylebox(ColorRole::PRIMARY, style_normal_color);
-	const Ref<StyleBoxFlat> button_hover = make_color_role_flat_stylebox(ColorRole::PRIMARY, style_hover_color);
-	const Ref<StyleBoxFlat> button_pressed = make_color_role_flat_stylebox(ColorRole::PRIMARY, style_pressed_color);
-	const Ref<StyleBoxFlat> button_disabled = make_color_role_flat_stylebox(ColorRole::PRIMARY, style_disabled_color);
-
-	// Ref<StyleBoxFlat> focus = make_flat_stylebox(style_focus_color, default_margin, default_margin, default_margin, default_margin, default_corner_radius, false, 2);
-	Ref<StyleBoxFlat> focus = make_color_role_flat_stylebox(ColorRole::PRIMARY, style_focus_color, default_margin, default_margin, default_margin, default_margin, default_corner_radius, false, 2);
+	const Ref<StyleBoxFlat> button_normal = make_color_role_flat_stylebox(ColorRole::PRIMARY, Color(1, 1, 1, 0.6));
+	const Ref<StyleBoxFlat> button_hover = make_color_role_flat_stylebox(ColorRole::PRIMARY, Color(1.5, 1.5, 1.5, 0.6));
+	const Ref<StyleBoxFlat> button_pressed = make_color_role_flat_stylebox(ColorRole::PRIMARY, Color(0.5, 0.5, 0.5, 0.6));
+	const Ref<StyleBoxFlat> button_disabled = make_color_role_flat_stylebox(ColorRole::PRIMARY, Color(1, 1, 1, 0.3));
+	Ref<StyleBoxFlat> focus = make_color_role_flat_stylebox(ColorRole::INVERSE_PRIMARY, Color(1, 1, 1, 0.75), default_margin, default_margin, default_margin, default_margin, default_corner_radius, false, 2);
 	// Make the focus outline appear to be flush with the buttons it's focusing.
 	focus->set_expand_margin_all(Math::round(2 * scale));
 
@@ -201,19 +192,20 @@ void fill_default_theme(Ref<Theme> &theme, const Ref<Font> &default_font, const
 
 	theme->set_color_scheme("default_color_scheme", "Button", default_color_scheme);
 
-	theme->set_color("font_color_scale", "Button", control_font_color);
-	theme->set_color("font_focus_color_scale", "Button", control_font_focus_color);
-	theme->set_color("font_pressed_color_scale", "Button", control_font_pressed_color);
-	theme->set_color("font_hover_color_scale", "Button", control_font_hover_color);
-	theme->set_color("font_hover_pressed_color_scale", "Button", control_font_pressed_color);
-	theme->set_color("font_disabled_color_scale", "Button", control_font_disabled_color);
+	theme->set_color("font_color_scale", "Button", Color(1, 1, 1, 1));
+	theme->set_color("font_focus_color_scale", "Button", Color(1.2, 1.2, 1.2, 1));
+	theme->set_color("font_pressed_color_scale", "Button", Color(1.3, 1.3, 1.3, 1));
+	theme->set_color("font_hover_color_scale", "Button", Color(1.2, 1.2, 1.2, 1));
+	theme->set_color("font_hover_pressed_color_scale", "Button", Color(1.3, 1.3, 1.3, 1));
+	theme->set_color("font_disabled_color_scale", "Button",  Color(1, 1, 1, 0.5));
 	theme->set_color("font_outline_color_scale", "Button", Color(1, 1, 1));
+
 	theme->set_color("icon_normal_color_scale", "Button", Color(1, 1, 1, 1));
-	theme->set_color("icon_focus_color_scale", "Button", Color(1, 1, 1, 1));
-	theme->set_color("icon_pressed_color_scale", "Button", Color(1, 1, 1, 1));
-	theme->set_color("icon_hover_color_scale", "Button", Color(1, 1, 1, 1));
-	theme->set_color("icon_hover_pressed_color_scale", "Button", Color(1, 1, 1, 1));
-	theme->set_color("icon_disabled_color_scale", "Button", Color(1, 1, 1, 0.4));
+	theme->set_color("icon_focus_color_scale", "Button", Color(1.2, 1.2, 1.2, 1));
+	theme->set_color("icon_pressed_color_scale", "Button", Color(1.3, 1.3, 1.3, 1));
+	theme->set_color("icon_hover_color_scale", "Button", Color(1.2, 1.2, 1.2, 1));
+	theme->set_color("icon_hover_pressed_color_scale", "Button", Color(1.3, 1.3, 1.3, 1));
+	theme->set_color("icon_disabled_color_scale", "Button", Color(1, 1, 1, 0.5));
 
 	theme->set_color("font_color", "Button", control_font_color);
 	theme->set_color("font_pressed_color", "Button", control_font_pressed_color);
@@ -222,7 +214,6 @@ void fill_default_theme(Ref<Theme> &theme, const Ref<Font> &default_font, const
 	theme->set_color("font_hover_pressed_color", "Button", control_font_pressed_color);
 	theme->set_color("font_disabled_color", "Button", control_font_disabled_color);
 	theme->set_color("font_outline_color", "Button", Color(1, 1, 1));
-
 	theme->set_color("icon_normal_color", "Button", Color(1, 1, 1, 1));
 	theme->set_color("icon_pressed_color", "Button", Color(1, 1, 1, 1));
 	theme->set_color("icon_hover_color", "Button", Color(1, 1, 1, 1));
@@ -276,12 +267,12 @@ void fill_default_theme(Ref<Theme> &theme, const Ref<Font> &default_font, const
 
 	theme->set_color_scheme("default_color_scheme", "MenuBar", default_color_scheme);
 
-	theme->set_color("font_color_scale", "MenuBar", control_font_color);
-	theme->set_color("font_pressed_color_scale", "MenuBar", control_font_pressed_color);
-	theme->set_color("font_hover_color_scale", "MenuBar", control_font_hover_color);
-	theme->set_color("font_focus_color_scale", "MenuBar", control_font_focus_color);
-	theme->set_color("font_hover_pressed_color_scale", "MenuBar", control_font_pressed_color);
-	theme->set_color("font_disabled_color_scale", "MenuBar", control_font_disabled_color);
+	theme->set_color("font_color_scale", "MenuBar", Color(1, 1, 1, 1));
+	theme->set_color("font_focus_color_scale", "MenuBar", Color(1.2, 1.2, 1.2, 1));
+	theme->set_color("font_pressed_color_scale", "MenuBar", Color(1.3, 1.3, 1.3, 1));
+	theme->set_color("font_hover_color_scale", "MenuBar", Color(1.2, 1.2, 1.2, 1));
+	theme->set_color("font_hover_pressed_color_scale", "MenuBar", Color(1.3, 1.3, 1.3, 1));
+	theme->set_color("font_disabled_color_scale", "MenuBar",  Color(1, 1, 1, 0.5));
 	theme->set_color("font_outline_color_scale", "MenuBar", Color(1, 1, 1));
 
 	theme->set_color("font_color", "MenuBar", control_font_color);
@@ -308,6 +299,7 @@ void fill_default_theme(Ref<Theme> &theme, const Ref<Font> &default_font, const
 	theme->set_color_role("font_disabled_color_role", "MenuBar", ColorRole::ON_PRIMARY);
 	theme->set_color_role("font_outline_color_role", "MenuBar", ColorRole::OUTLINE);
 
+
 	theme->set_constant("h_separation", "MenuBar", Math::round(4 * scale));
 
 	// LinkButton
