From 911562835a79df9dc99c1d049fbca2077b352464 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=9B=85=E8=A7=85=E5=A8=B1=5C=E7=93=B6=E5=AD=90?=
 <pingzi@yamiyu.com>
Date: Fri, 20 Oct 2023 09:37:50 +0800
Subject: [PATCH] =?UTF-8?q?=E5=90=88=E5=B9=B6?=
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

---
 editor/editor_help.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/editor/editor_help.h b/editor/editor_help.h
index 1f813f930c675b5b6dc5755d641da255a579f6a5..8fa38866dfb29778b8f76e72e5b08455c5dccd2b 100644
--- a/editor/editor_help.h
+++ b/editor/editor_help.h
@@ -177,7 +177,9 @@ class EditorHelp : public VBoxContainer {
 
 	Error _goto_desc(const String &p_class);
 	//void _update_history_buttons();
+
 	void _update_method_list(const Vector<DocData::MethodDoc> p_methods, MethodType p_method_type);
+
 	void _update_method_descriptions(const DocData::ClassDoc p_classdoc, const Vector<DocData::MethodDoc> p_methods, MethodType p_method_type);
 	void _update_doc();
 
