From ed28e352ea79f74305103cc045549e393e0605cc Mon Sep 17 00:00:00 2001
From: aiaimimi0920 <153103332+aiaimimi0920@users.noreply.github.com>
Date: Mon, 4 Mar 2024 09:25:53 +0800
Subject: [PATCH] # cache

---
 scene/gui/control.cpp | 23 ++++++++++++-----------
 1 file changed, 12 insertions(+), 11 deletions(-)

diff --git a/scene/gui/control.cpp b/scene/gui/control.cpp
index e6bd537dfecdaaf321ec334205e75aebd595d88e..63d0db83991f5661350c9f0a0b865e949a8f3426 100644
--- a/scene/gui/control.cpp
+++ b/scene/gui/control.cpp
@@ -216,8 +216,8 @@ void Control::get_argument_options(const StringName &p_function, int p_idx, List
 			ThemeDB::get_singleton()->get_default_theme()->get_constant_list(get_class(), &sn);
 		} else if (pf == "add_theme_color_role_override" || pf == "has_theme_color_role" || pf == "has_theme_color_role_override" || pf == "get_theme_color_role") {
             ThemeDB::get_singleton()->get_default_theme()->get_color_role_list(get_class(), &sn);
-        // } else if (pf == "add_theme_color_scheme_override" || pf == "has_theme_color_scheme" || pf == "has_theme_color_scheme_override" || pf == "get_theme_color_scheme") {
-        //     ThemeDB::get_singleton()->get_default_theme()->get_color_scheme_list(get_class(), &sn);
+        } else if (pf == "add_theme_color_scheme_override" || pf == "has_theme_color_scheme" || pf == "has_theme_color_scheme_override" || pf == "get_theme_color_scheme") {
+            ThemeDB::get_singleton()->get_default_theme()->get_color_scheme_list(get_class(), &sn);
 		}
 		sn.sort_custom<StringName::AlphCompare>();
 		for (const StringName &name : sn) {
@@ -308,10 +308,10 @@ bool Control::_set(const StringName &p_name, const Variant &p_value) {
             String dname = name.get_slicec('/', 1);
             data.theme_color_role_override.erase(dname);
             _notify_theme_override_changed();
-        // } else if (name.begins_with("theme_override_color_schemes/")) {
-        //     String dname = name.get_slicec('/', 1);
-        //     data.theme_color_scheme_override.erase(dname);
-        //     _notify_theme_override_changed();
+        } else if (name.begins_with("theme_override_color_schemes/")) {
+            String dname = name.get_slicec('/', 1);
+            data.theme_color_scheme_override.erase(dname);
+            _notify_theme_override_changed();
 		} else {
 			return false;
 		}
@@ -338,9 +338,9 @@ bool Control::_set(const StringName &p_name, const Variant &p_value) {
 		} else if (name.begins_with("theme_override_color_roles/")) {
             String dname = name.get_slicec('/', 1);
             add_theme_color_role_override(dname, p_value);
-        // } else if (name.begins_with("theme_override_color_schemes/")) {
-        //     String dname = name.get_slicec('/', 1);
-        //     add_theme_color_scheme_override(dname, p_value);
+        } else if (name.begins_with("theme_override_color_schemes/")) {
+            String dname = name.get_slicec('/', 1);
+            add_theme_color_scheme_override(dname, p_value);
 		} else {
 			return false;
 		}
@@ -2777,6 +2777,7 @@ Ref<ColorScheme> Control::get_theme_color_scheme(const StringName &p_name, const
     if (!data.initialized) {
         WARN_PRINT_ONCE(vformat("Attempting to access theme items too early in %s; prefer NOTIFICATION_POSTINITIALIZE and NOTIFICATION_THEME_CHANGED", this->get_description()));
     }
+	return Ref<ColorScheme>();
 
     if (p_theme_type == StringName() || p_theme_type == get_class_name() || p_theme_type == data.theme_type_variation) {
         const Ref<ColorScheme> *color_scheme = data.theme_color_scheme_override.getptr(p_name);
@@ -2813,8 +2814,8 @@ Variant Control::get_theme_item(Theme::DataType p_data_type, const StringName &p
 			return get_theme_stylebox(p_name, p_theme_type);
 		case Theme::DATA_TYPE_COLOR_ROLE:
             return get_theme_color_role(p_name, p_theme_type);
-        // case Theme::DATA_TYPE_COLOR_SCHEME:
-        //     return get_theme_color_scheme(p_name, p_theme_type);
+        case Theme::DATA_TYPE_COLOR_SCHEME:
+            return get_theme_color_scheme(p_name, p_theme_type);
 		case Theme::DATA_TYPE_MAX:
 			break; // Can't happen, but silences warning.
 	}
